xcode/Subconscious/Shared/Components/Common/Profile/UserProfileView.swift

```diff
@@ -143,11 +143,13 @@ struct UserProfileView: View {
                             onProfileAction(user, action)
                         },
                         onTapStatistics: {
-                            send(
-                                .tabIndexSelected(
-                                    UserProfileDetailModel.followingTabIndex
+                            withAnimation {
+                                send(
```

Comment:
Use `.animation()` modifier on `Update` in `.tabSelected` handler instead.

Why:

- More consistent. Keeps logic in update function
- More control over animations.
    - You can trigger an effect which triggers an animation later on.
    - Can drive animation as a result of model state
- Store joins actions on main thread, to ensure that results of effects never change the UI off main thread (disallowed). This can mean some state changes can happen "next tick". `withAnimation` sets a global state during the same tick that `send` is called, but by the time the state changes, it might be next tick, so you can lose the animation in some cases.

`Update(state: state).animation(.default)` solves all of these issues.